metricbeat/module/prometheus/collector/data.go

```diff
@@ -59,7 +62,7 @@ func getPromEventsFromMetricFamily(mf *dto.MetricFamily) []PromEvent {
 			if !math.IsNaN(counter.GetValue()) && !math.IsInf(counter.GetValue(), 0) {
 				events = append(events, PromEvent{
 					data: common.MapStr{
-						name: counter.GetValue(),
+						name: rateFloat64(ccache, name, labels, counter.GetValue()),
```

Comment:
nit: this is only returning rate if ccache is not nil.

while I think the code is understandable, in my opinionated self I would have a check after the `if` above that does

```
value = counter.GetValue()
if ccache != nil {
  value = rateFloat64(ccache, name, labels, value)
}
```

to make clear that we are not rating always.

Reply:
I tried to avoid that as this is done for many values in this block. That say I agree this can be misleading. What about renaming `rateFloat64` to something like `getCounterOrRate`?

Reply:
or RateIfEnabled :innocent:

Reply:
I noticed after continuing reading.

The name change is ok from my POV, but not _very_ needed. I think the code is very self-explanatory.